bundles/framework/admin-users/view/UserForm.jsx

```diff
@@ -0,0 +1,154 @@
+                <StyledInput
+                    className='t_email'
+                    value={state.userFormState.email}
+                    onChange={(e) => controller.updateUserFormState('email', e.target.value)}
+                    status={state.userFormErrors.includes('email') ? 'error' : null}
+                    disabled={isExternal}
+                />
+            </LabelledField>
+            {!isExternal && (
+                <>
+                    <LabelledField>
+                        <Label><Message messageKey='flyout.adminusers.pass' /> {!state.editingUserId && (getMandatoryIcon(true, state.userFormState.password))}</Label>
+                        <StyledInput
+                            className='t_password'
+                            type='password'
+                            value={state.userFormState.password}
+                            onChange={(e) => controller.updateUserFormState('password', e.target.value)}
+                            status={state.userFormErrors.includes('password') ? 'error' : null}
+                        />
+                    </LabelledField>
+                    <LabelledField>
+                        <Label><Message messageKey='flyout.adminusers.pass_retype' /> {!state.editingUserId && (getMandatoryIcon(true, state.userFormState.rePassword))}</Label>
+                        <StyledInput
+                            className='t_re_password'
+                            type='password'
+                            value={state.userFormState.rePassword}
+                            onChange={(e) => controller.updateUserFormState('rePassword', e.target.value)}
+                            status={state.userFormErrors.includes('password') ? 'error' : null}
+                        />
+                    </LabelledField>
+                </>
+            )}
+            <LabelledField>
+                <Label><Message messageKey='flyout.adminusers.addRole' /></Label>
+                <StyledSelect
+                    className='t_roles'
+                    mode='multiple'
+                    allowClear
+                    onChange={(value) => controller.updateUserFormState('roles', value)}
+                    defaultValue={state.userFormState.roles}
+                    status={state.userFormErrors.includes('roles') ? 'error' : null}
+                    options={state.roles.map(role => (
+                        {
+                            label: role.name,
+                            value: role.id
+                        }
+                    ))}
+                />
+            </LabelledField>
+            <Buttons>
+                <RightButtons>
+                    <SaveButton
+                        type='save'
+                        onClick={() => controller.saveUser()}
+                    />
+                    {(state.editingUserId && !isExternal) && (
+                        <Confirm
```

Comment:
Let's use the common delete button in oskari-ui here as well if there's not some reason that we can't.